File: 이것이 코딩테스트다/05.이진 탐색/2.재귀함수로 구현한 이진탐색.js
const texts = [10, 7, [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]];

const n = texts.shift();
const target = texts.shift();
const array = texts.pop();

const binarySearch = (array, target, start, end) => {
  if (start > end) {
    return NaN;
  }
  mid = Math.floor((start + end) / 2);
  if (array[mid] === target) {
    return mid + 1;
  } else if (array[mid] < target) {
    return binarySearch(array, target, mid + 1, end);
  } else {
    return binarySearch(array, target, start, mid - 1);
  }
};
console.log(binarySearch(array, target, 0, n));

File: 이것이 코딩테스트다/05.이진 탐색/2.재귀함수로 구현한 이진탐색.py
def binary_search (array, target, start, end):
  if start > end:
    return None
  mid = (start + end) // 2
  if target < array[mid]:
    return binary_search(array, target, start, mid-1)
  elif target > array[mid]:
    return binary_search(array, target, mid+1, end)
  else:
    return mid+1

n, target = map(int, input().split())
array = list(map(int, input().split()))

print(binary_search(array, 0, 0, n))